tests/ui/unnecessary_to_owned.stderr

```diff
@@ -589,5 +589,11 @@ error: unnecessary use of `to_vec`
 LL |     s.remove(&(&["b"]).to_vec());
    |              ^^^^^^^^^^^^^^^^^^ help: replace it with: `(&["b"]).as_slice()`
 
-error: aborting due to 88 previous errors
+error: unnecessary use of `into_owned.into_iter()`
+  --> tests/ui/unnecessary_to_owned.rs:594:5
+   |
+LL |     cow.into_owned().into_iter()
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `cow.into_owned()`
```

Comment:
The suggestion works in this specific case, but it doesn't seem like it would work in the general case if the iterator is later used as part of another expression, e.g. for `cow.into_owned().into_iter().map(...).collect()` the `.into_iter()` is required. Tbh I don't think we can really suggest anything here for `Cow::into_owned` and we probably should just not lint at all here?

Reply:
Thank you for the review! I'll make a change.

Reply:
Seems to be fixed in https://github.com/rust-lang/rust-clippy/pull/13853/commits/08d8c4a67fd9a9eea1bad23da92bb614bc6f9966.